src/lib/components/fileUploader/fileUploader.js

```diff
@@ -0,0 +1,234 @@
+    if (hasErrorAfterSubmit !== prevProps.hasErrorAfterSubmit) {
+      const validationErrors = checkValidations(validations, filesUploaded);
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ validationErrors });
+    }
+  }
+
+  getFilesUploadedAsArray = props => {
+    if (props.filesUploaded && props.filesUploaded.length) {
+      return props.filesUploaded.split(", ");
+    } else {
+      return [];
+    }
+  };
+
+  handleUploadFiles = (file, data) => {
+    const {
+      name,
+      filesUploaded,
+      onFilesUpload,
+      allowMultipleUploads,
+    } = this.props;
+
+    if (!file && !data) {
+      onFilesUpload(name, "");
+      return;
+    }
+    let dataToUpdate;
+    if (!filesUploaded) {
+      dataToUpdate = data.uploadURL;
+    } else if (allowMultipleUploads) {
+      dataToUpdate = `${filesUploaded}, ${data.uploadURL}`;
+    }
+    this.setState({ validationErrors: "" });
+    onFilesUpload(name, dataToUpdate);
+  };
+
+  handleDeleteFile = fileToDelete => e => {
+    e.preventDefault();
+    const { name, allowMultipleUploads, onFilesUpload } = this.props;
+    if (!allowMultipleUploads) {
+      onFilesUpload(name, "");
+    } else {
+      const filesArray = this.getFilesUploadedAsArray(this.props);
+      const newFilesArray = filesArray.filter(file => file !== fileToDelete);
+      onFilesUpload(name, newFilesArray.join(", "));
+    }
+    this.uppy.getFiles().forEach(file => {
+      if (file.name === fileToDelete || file.uploadURL === fileToDelete) {
+        this.uppy.removeFile(file.id);
+      }
+    });
+  };
+
+  handleOpen = () => this.setState({ modalOpen: true });
+
+  handleClose = () => this.setState({ modalOpen: false });
+
+  render() {
+    const { isDashboard, height, webcam, placeholder, theme } = this.props;
+    const filesUploadedArray = this.getFilesUploadedAsArray(this.props);
+    const { validationErrors } = this.state;
+
+    return (
+      <div
+        className={`${styles.uppyUploadContainer} ${styles[`theme-${theme}`]}`}
+      >
+        {isDashboard ? (
+          <Dashboard
+            note={`When you're done, click the x in the
+            top right to return to the form.`}
+            uppy={this.uppy}
+            plugins={webcam ? ["s3", "Webcam"] : ["s3"]}
+            height={height}
+          />
+        ) : (
+          <React.Fragment>
+            <button
+              className={`${styles.documentItemButton} ${validationErrors &&
+                styles.withErrors}`}
+              onClick={this.handleOpen}
+              type="button"
+            >
+              {placeholder || "Upload Files"}
+            </button>
+            <DashboardModal
+              uppy={this.uppy}
+              plugins={webcam ? ["s3", "Webcam"] : ["s3"]}
+              closeModalOnClickOutside
+              open={this.state.modalOpen}
+              onRequestClose={this.handleClose}
+              note={`When you're done, click the x in the
```

Comment:
Is there a way to make the `x` larger i had trouble finding it haha maybe just my eyes. 
Also an edge use-case so not major right now, but I think there is a build in option to press `esc` to navigate away from the upload modal /close modal, right now for me that doesn't work, it just reveals the scroll overflow, and then when I try to click the `x` it doesn't work.

Reply:
I couldn't see the closing `x` at all... But maybe I was looking at a diff thing to you?

Reply:
it is just to the right of the drag and drop area, it looks like its behind modal.